### PR TITLE
Add compare_communities unit tests

### PR DESCRIPTION
**Pull request by GroteGnoom**
Part of #1592

The Adjusted Rand on 2 vertices in 2 clusters still outputs -NaN,
but that seems correct and a bit excessive to check for up front.

I'm not sure why this function is used instead of just having separate
function calls. Separate function calls would also clear up the
documentation.

### CODE REVIEW

**Comment by GroteGnoom** [comment]
Allright, linux outputs:
`Adjusted Rand result: -nan`
and Windows and Mac do:
`Adjusted Rand result: -nan(ind)`
I'll see what's going on there.

**Comment by ntamas** [comment]
We have `igraph_real_printf()` in `igraph_types.h` that is meant to work around platform-specific printing styles for NaN. We also have `print_real()` in `test_utilities.inc` for the same purpose.

**Comment by GroteGnoom** [comment]
Ah, perfect, thank you!

File: src/community/misc.c [hunk 1]
```diff
@@ -703,6 +725,11 @@ static int igraph_i_compare_communities_rand(
     double rand, n;
     double frac_pairs_in_1, frac_pairs_in_2;
 
+    if (igraph_vector_size(v1) <= 1) {
+        IGRAPH_ERRORF("Rand indices not defined for only zero or one vertices. "
+        "Found memberschip vector of size %ld", IGRAPH_EINVAL, igraph_vector_size(v1));
```
Comment (ntamas) [inline src/community/misc.c:730]: Typo here :) (`memberschip`)

File: tests/unit/igraph_compare_communities.c [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include <igraph.h>
+#include "test_utilities.inc"
+
+char *names[] = {
+    [IGRAPH_COMMCMP_VI] = "VI",
```
Comment (ntamas) [inline tests/unit/igraph_compare_communities.c:23]: Hmmm, this is nice but I wonder whether this construct is understood by most C compilers. Is this a standard thing? @szhorvat ?

**Comment by szhorvat** [comment]
All is fine with this PR, but I'm confused about how one gets a `-NaN`. I thought NaNs don't have a sign. Perhaps some implementations print the sign bit anyway?

**Comment by ntamas** [comment]
Are we using `-ffast-math`? I think that NaNs get all screwed up with that.

File: tests/unit/igraph_compare_communities.c (hunk 2 above)
Comment (GroteGnoom) [reply tests/unit/igraph_compare_communities.c:23]: Designated initializers are in C99, not in earlier standards. I hope I remember correctly we use C99.

Figuring out if MSVC actually supports it is hard, but I found a blog post comment which references another comment that doesn't exist anymore which seems to indicate that VS 2013 supports it.

https://devblogs.microsoft.com/cppblog/c11-and-c17-standard-support-arriving-in-msvc/#comment-1230

I used it to not depend on the enum order.

**Comment by GroteGnoom** [comment]
Thanks for that website 🙂 It's kind of strange how complicated a NaN is. 

I do agree that -NaN doesn't make sense, but I guess in practice (as in this case) it just means -0.0/0.0.

File: tests/unit/igraph_compare_communities.c (hunk 2 above)
Comment (szhorvat) [reply tests/unit/igraph_compare_communities.c:23]: I tried it here with VS2015, seems to be good: https://godbolt.org/z/z3qWa1

Comment (szhorvat) [reply tests/unit/igraph_compare_communities.c:23]: Note that it's only available in C mode, not in C++ mode, but that's fine.

We should keep supporting VS2015, i.e. version 19.0 (not 19.x where x > 0), for the time being. This seems to be fine with 19.0

**Pull request merged**